### PR TITLE
refactor specs

### PR DESCRIPTION
**Pull request by xithan**
includes also an overhaul of seed.rb: 
https://github.com/decko-commons/decko-seed/commit/dd314d1ea12af2ea2e84959a73c3143762d6b14f#diff-8cc501c146c47a19753410dbcd5e21d3

### CODE REVIEW

File: card/mod/core/spec/set/all/assign_attributes_spec.rb [hunk 1]
```diff
@@ -1,314 +1,37 @@
...
-    assert_equal new_name, card.name
-    assert Card[new_name]
-  end
-
-  def card name
-    Card[name].refresh || raise("Couldn't find card named #{name}")
-  end
-end
-
 RSpec.describe Card::Set::All::AssignAttributes do
```
Comment (xithan) [inline card/mod/core/spec/set/all/assign_attributes_spec.rb]: All the tests below have nothing to do with `Card::Set::All::AssignAttributes`.
Some of them test the same behaviour that is already tested in `fetch_spec.rb`. 
I removed them and refactored the other and moved them to the right place.

File: card/mod/account/spec/set/all/account_spec.rb [hunk 2]
```diff
@@ -1,140 +1,133 @@
...
       expect(Card["A"].accountable?).to eq(false)
     end
 
-    it "is true for cards with *accountable rule on" do
-      Card::Auth.as_bot do
-        Card.create name: "A+*self+*accountable", content: "1"
-        Card.create name: "*account+*right+*create",
-                    content: "[[Anyone Signed In]]"
-      end
+    it "is true for cards with *accountable rule on", as_bot: true do
```
Comment (xithan) [inline card/mod/account/spec/set/all/account_spec.rb]: new test api to get rid of all these `Card::Auth.as_bot` block in our test

File: card/mod/core/set/all/name.rb [hunk 3]
```diff
@@ -240,36 +240,9 @@ def repair_key
...
-    Card.where(id: de.id).update_all name: newname.to_s, key: newname.key
-    de.update_referers = update_referers
-    de.refresh_references_in
-    Card.expire newname
-  end
+event :name_update_finished, :finalize, changed: :name, on: :save do
+  # The events to update references has to happen after :cascade_name_changes,
+  # but :cascade_name_changes is defined after the reference events and
+  # and additionaly it is defined on :update but some of the reference
+  # events are on :save. Hence we need this additional hook to organize these events.
```
Comment (xithan) [inline card/mod/core/set/all/name.rb]: !

File: card/mod/account/spec/set/all/account_spec.rb (hunk 2 above)
Comment (ethn) [reply card/mod/account/spec/set/all/account_spec.rb]: nice!

File: card/mod/account/spec/set/all/account_spec.rb [hunk 4]
```diff
@@ -1,139 +1,132 @@
...
-    end
-
-    context "for Joe User" do
-      before do
-        @joe_user_card = Card::Auth.current
-        @parties = @joe_user_card.parties # note: must be called to test resets
+    describe "parties" do
+      it "for Wagn Bot", with_user: Card::WagnBotID do
+        expect(Card::Auth.current.parties.sort).to eq(
+                                                     [Card::WagnBotID, Card::AnyoneSignedInID, Card::AdministratorID]
```
Comment (ethn) [inline card/mod/account/spec/set/all/account_spec.rb]: not clear on our new line length policy, but this wraps on me at github with the browser at full screen (in split diff mode).

File: card/mod/admin/set/self/admin.rb [hunk 5]
```diff
@@ -2,6 +2,16 @@
...
 
+def run_task_from_task_basket task
+  task = task.to_sym
+  task_data = tasks.find {|h| h[:name].to_sym == task.to_sym}
+  if !irreversibles_tasks_allowed? && task_data[:irreversible]
+    not_allowed task_data[:stats][:link_text]
+  else
+    task_data[:execute_policy].call if task_data
+  end
+end
```
Comment (ethn) [inline card/mod/admin/set/self/admin.rb]: more readable.  thanks.

File: card/mod/account/spec/set/all/account_spec.rb [hunk 6]
```diff
@@ -1,139 +1,132 @@
...
+          # local cache still has old parties
+          # (permission does not change mid-request)
+
+          Card::Cache.restore
+          # simulate new request
+          # clears local cache, where, eg, @parties would still be cached on card
+
+          Card::Auth.current_id = Card::Auth.current_id
+          # simulate new request
+          # current_id assignment clears several class variables
```
Comment (ethn) [inline card/mod/account/spec/set/all/account_spec.rb]: kinda seems like simulate_new_request might be a useful method...

File: card/mod/core/set/all/name.rb (hunk 3 above)
Comment (ethn) [reply card/mod/core/set/all/name.rb]: woof.  so there are no known bugs associated with this?

Was the thinking before that we could figure out all the reference changes from parts?

File: card/mod/core/spec/set/all/assign_attributes_spec.rb (hunk 1 above)
Comment (ethn) [reply card/mod/core/spec/set/all/assign_attributes_spec.rb]: thank you.

**Pull request merged**